src/sql/workbench/contrib/notebook/test/browser/notebookMarkdown.test.ts

```diff
@@ -85,6 +85,13 @@ suite('NotebookMarkdownRenderer', () => {
 		assert.strictEqual(result.innerHTML, `<p><a href="https://www.test.com?test=&amp;test2=" data-href="https://www.test.com?test=&amp;test2=" title="https://www.test.com?test=&amp;test2=" is-markdown="true" is-absolute="false">test</a></p>`);
 	});
 
+	test('link to section in the same file', () => {
+		let result: HTMLElement = notebookMarkdownRenderer.renderMarkdown({ value: `[#section](#section)`, isTrusted: true });
```

Comment:
How about removing the # on the **section** in square brackets? That way we can be sure that the link and the displayed text aren't getting mixed up in the rendered results.